public/js/reducers/sources.js

```diff
@@ -13,15 +13,13 @@ import type { Record } from "../utils/makeRecord";
 export type SourcesState = {
   sources: I.Map<string, any>,
   selectedSource: ?any,
-  sourcesText: I.Map<string, any>,
   tabs: I.List<any>,
   sourceMaps: I.Map<string, any>,
 };
 
 const State = makeRecord(({
   sources: I.Map(),
   selectedSource: undefined,
-  sourcesText: I.Map(),
```

Comment:
Haven't gone through all of this PR yet, but I will tomorrow. Thanks for pointing this out, because I think you already know that I think it should stay separate :)

This is a case where I want the data to be more denormalized, like a separate table in a db. A small nit that now the property is `sources.text.text`. My bigger complaint is that now the sources are not the canonical "protocol objects" as I call them: objects that are given to use from the client. The nice thing about that is that it's easy to back and forth. For example, in my WIP PR I snapshotted the source objects into this file: https://github.com/jlongster/debugger.html/blob/4b79f1f14ead1620fe01dbb549920d891d9f4246/public/js/test/fixtures/todoSources.json With those, I can pick them out and immediately send them into the `newSource` action. Changing the structure means I can't always go back and forth between actions and selectors.

I would like for the same constraint to be held to all protocol objects stored in the reducer, for example the `loadedObjects` field. It makes it easy to reconstruct state and work with the data in all of our testing infrastructure.

I'd like to hear more about your "pretty print explorations", or maybe I will see them when I look closer at this PR. We can write code to "convert" this state back into protocol objects when we need it, but I'd like to see what this actually made simpler.